services/eth_gateway.go

```diff
@@ -266,6 +266,12 @@ func getCurrentBlock() (*types.Block, error) {
 // WaitForTransfer is blocking call that will observe on brokerAddr on transfer on ETH.
 // If it is completed return number of PRL.
 func waitForTransfer(brokerAddr common.Address) (*big.Int, error) {
+	balance := checkBalance(brokerAddr)
+	if balance.Int64() > 0 {
+		// Has balance already, don't need to wait for it.
+		return balance, nil
+	}
```

Comment:
Does checkBalance return the balance of PRL or the balance of ETH?  Or both?

Reply:
Ideally, it should be PRL, and we use it in the codebase. Maybe Astor could have a better answer

Reply:
@Astor ^

Reply:
Ultimately we will need methods to get balances of both ETH and PRL. I created a Zenhub issue to have methods for both.   

If everyone is cool with it we could just go ahead and approve this, and splitting out those methods could be something we do soonish. We will have to look through the codebase at where the current method is being used and make sure it is replaced with the correct one.

Reply:
It checks the for both balances.